invisible_cities/cities/penthesilea.py

```diff
@@ -72,48 +72,58 @@ def _file_loop(self, write_hits):
         return nevt_in, nevt_out
 
     def _event_loop(self, event_numbers, timestamps,
-                    nevt_in, nevt_out, write_hits, S1s, S2s, S2Sis):
+                    nevt_in, nevt_out, write_hits, S1_dict, S2_dict, S2Si_dict):
         max_events_reached = False
         for evt_number, evt_time in zip(event_numbers, timestamps):
             nevt_in += 1
             if self.max_events_reached(nevt_in):
                 max_events_reached = True
                 break
-            S1   = S1s  .get(evt_number, {})
-            S2   = S2s  .get(evt_number, {})
-            S2Si = S2Sis.get(evt_number, {})
+            s1   = S1_dict  .get(evt_number, {})
+            s2   = S2_dict  .get(evt_number, {})
+            s2si = S2Si_dict.get(evt_number, {})
 
-            f1 = s1s2_filter(self._s1s2_selector, S1, S2, S2Si)
-            f2 = s2si_filter(S2Si)
+            f1 = s1s2_filter(self._s1s2_selector, s1, s2, s2si)
+            f2 = s2si_filter(s2si)
             if not f1 or not f2:
                 continue
             nevt_out += 1
 
-            evt = self._create_hits_event(evt_number, evt_time, S1, S2, S2Si)
+            evt = self._create_hits_event(evt_number, evt_time, s1, s2, s2si)
             write_hits(evt)
 
             self.conditional_print(evt, nevt_in)
 
         return nevt_in, nevt_out, max_events_reached
 
-    def _create_hits_event(self, evt_number, evt_time, S1, S2, Si):
+    def _create_hits_event(self, evt_number, evt_time,
+                           s1, s2, s2si):
 
         hitc = PersistentHitCollection(evt_number, evt_time * 1e-3)
         #hitc.evt   = evt_number
         #hitc.time  = evt_time * 1e-3 # s
 
-        t, e = next(iter(S1.values()))
-        S1t  = t[np.argmax(e)]
+        # in order to compute z one needs to define one S1
+        # for time reference. By default the filter will only
+        # take events with exactly one s1. Otherwise, the
+        # convention is to take the first peak in the S1 object
+        # as reference.
 
-        S2, Si = self.rebin_s2(S2, Si)
+        # peak  = S1.peak_waveform(0)
+        # t, e = next(iter(S1.values()))
+        # S1t  = t[np.argmax(e)]
+
+        s1_t = s1.peak_waveform(0).tpeak
+
+        #S2, Si = self.rebin_s2(S2, Si) TODO REVISE
```

Comment:
Again, I think this is the same comment: rebin_s2 calls cpf.rebin_S2 which has not existed in our master repository for months.  

If the idea is that penthesilea wants to rebin the s2 and s2si such that they have time bins larger than those saved in the pmaps, we can write a tiny function to do that. It should be very simple since now the `rebin_waveform` function returns clean usually 1 mus time bins. If not, we can just delete this.